src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/jvmsizing/OldGenReclamationRca.java

```diff
@@ -0,0 +1,147 @@
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics.GCType.OLD_GEN;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics.GCType.TOT_FULL_GC;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics.HeapDimension.MEM_TYPE;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metricsdb.MetricsDB;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Metric;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Rca;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Resources.State;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators.SlidingWindow;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators.SlidingWindowData;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.contexts.ResourceContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.ResourceFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.persist.SQLParsingUtil;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotResourceSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.ResourceUtil;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.hotheap.HighHeapUsageOldGenRca.MinOldGenSlidingWindow;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class OldGenReclamationRca extends Rca<ResourceFlowUnit<HotResourceSummary>> {
+
+  private static final long EVAL_INTERVAL_IN_S = 5;
+  private static final double DEFAULT_TARGET_UTILIZATION_AFTER_GC = 75.0d;
+  private static final long DEFAULT_RCA_EVALUATION_INTERVAL_IN_S = 60;
+  private static final long B_TO_MB = 1024 * 1024;
+
+  private final MinOldGenSlidingWindow minOldGenSlidingWindow;
+  private final SlidingWindow<SlidingWindowData> gcEventsSlidingWindow;
+
+  private Metric heapUsed;
+  private Metric gcEvent;
+  private Metric heapMax;
+  private HotResourceSummary prevSummary;
+  private ResourceContext prevContext;
+  private double targetHeapUtilizationAfterGc;
+  private long rcaEvaluationIntervalInS;
+  private long rcaPeriod;
+  private int samples;
+
+  public OldGenReclamationRca(final Metric heapUsed, final Metric heapMax, final Metric gcEvent) {
+    this(heapUsed, heapMax, gcEvent, DEFAULT_TARGET_UTILIZATION_AFTER_GC,
+        DEFAULT_RCA_EVALUATION_INTERVAL_IN_S);
+  }
+
+  public OldGenReclamationRca(final Metric heapUsed, final Metric heapMax, final Metric gcEvent,
```

Comment:
Isn't the HighHeapUsageOldGenRca doing something similar with `minOldGenSlidingWindow` ? can we reuse flow units from there?

Reply:
They're similar, but I decided to keep them separate in case one needs to change independently of the other. I've reduced duplication though. Added a base oldgen class that contains common methods used by old gen rcas, like using minOldGenSlidingWindow to get heap usage after gc, and getting max(and used) size for old gen.